Dockerfile

```diff
@@ -1,11 +1,25 @@
-FROM rust:1.34.0-slim
+FROM debian:buster-slim
 
 RUN apt-get update
 RUN apt-get install -y clang cmake
 RUN apt-get install -y libsnappy-dev
+RUN apt-get install -y curl
 
 RUN adduser --disabled-login --system --shell /bin/false --uid 1000 user
 
+ARG RUST_VERSION=1.34.0
+ENV RUSTUP_HOME /usr/local/rustup
+ENV CARGO_HOME /usr/local/cargo
+ENV PATH $CARGO_HOME/bin:$PATH
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- \
+  -y \
+  --verbose \
+  --profile minimal \
+  --default-toolchain $RUST_VERSION
```

Comment:
I meant to install it using `apt-get` to get signature validation for free. If you prefer rustup, then I'd suggest doing what the official Docker image does and check the hashes.

Reply:
Good point regarding `apt` over `rustup`.

However regarding rustup, it doesn't seem possible to get a secure Rust environment via `rustup` which was why I intentionally didn't bother with hashes.

For example the official Docker image only checks the hash of `rustup-init.sh`. This is pointless because:
1. `rustup-init.sh` then downloads `rustup` but doesn't check hashes/sigs.
2. `rustup` doesn't check signatures when installing Rust versions.

Se verifying `rustup-init.sh` doesn't guarantee anything about `rustup`, and even if you get an honest `rustup`, that doesn't even guarantee you get an honest `rustc`/`cargo`!

But you're right about preferring `apt`, that would definitely be better.